runtime/shibuya/src/xcm_config.rs

```diff
@@ -66,27 +67,34 @@ pub type CurrencyTransactor = CurrencyAdapter<
     (),
 >;
 
-pub struct AsAssetWithRelay<AssetId, GeneralAssetConverter>(
-    sp_std::marker::PhantomData<(AssetId, GeneralAssetConverter)>,
+pub struct AssetLocationIdConverter<AssetId, AssetMapper>(
+    sp_std::marker::PhantomData<(AssetId, AssetMapper)>,
 );
-impl<AssetId, GeneralAssetConverter> xcm_executor::traits::Convert<MultiLocation, AssetId>
-    for AsAssetWithRelay<AssetId, GeneralAssetConverter>
+impl<AssetId, AssetMapper> xcm_executor::traits::Convert<MultiLocation, AssetId>
+    for AssetLocationIdConverter<AssetId, AssetMapper>
 where
     AssetId: Clone + Eq + Bounded,
-    GeneralAssetConverter: xcm_executor::traits::Convert<MultiLocation, AssetId>,
+    AssetMapper: AssetLocationGetter<AssetId>,
 {
-    fn convert_ref(id: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
-        if id.borrow().eq(&MultiLocation::parent()) {
+    fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
+        // TODO should we remove this hardcoding completely?
+        if location.borrow().eq(&MultiLocation::parent()) {
             Ok(AssetId::max_value())
```

Comment:
@akru I want to remove this and just use the new `AssetMapper`.
It's generic and relay chain asset is same as any other so I don't see reason to have this hardcoded.

Do you agree?

Reply:
Absolutely, let’s make it generic